examples/Dicom/src/parseDicomFiles.js

```diff
@@ -0,0 +1,474 @@
+    return 'StudyInstanceUID'
+  }
+
+  static get tags() {
+    return [
+      'StudyID',
+      'StudyInstanceUID',
+      'StudyDate',
+      'StudyTime',
+      'AccessionNumber',
+      'StudyDescription',
+      ]
+    }
+
+  constructor(metaData, file) {
+    super()
+    this.series = {}
+    this.extractTags(metaData)
+    this.addMetaData(metaData, file)
+  }
+
+  addMetaData(metaData, file) {
+    const serieNumber = metaData[DICOMSeries.primaryTag]
+    let serie = this.series[serieNumber]
+    if (serie === undefined) {
+      serie = new DICOMSeries(metaData, file)
+      this.series[serieNumber] = serie
+    } else {
+      serie.addMetaData(metaData, file)
+    }
+  }
+}
+
+class DICOMSeries extends DICOMEntity {
+  static get primaryTag() {
+    return 'SeriesInstanceUID'
+  }
+
+  static get tags() {
+    return [
+      'SeriesNumber',
+      'SeriesInstanceUID',
+      'SeriesDate',
+      'SeriesTime',
+      'Modality',
+      'SeriesDescription',
+      'ProtocolName',
+      'FrameOfReferenceUID',
+      'TransferSyntaxUID',
+      ]
+    }
+
+  constructor(metaData, file) {
+    super()
+    this.images = {}
+    this.extractTags(metaData)
+    this.addMetaData(metaData, file)
+  }
+
+  addMetaData(metaData, file) {
+    const imageNumber = metaData[DICOMImage.primaryTag]
+    if (imageNumber in this.images) {
+      const id = metaData[DICOMSeries.primaryTag]
+      throw Error(`Image ${imageNumber} already added to serie ${id}.`)
+    }
+    this.images[imageNumber] = new DICOMImage(metaData, file)
+  }
+
+  getImageData() {
+    function numArrayFromString(str, separator = '\\') {
+      const strArray = str.split(separator)
+      return strArray.map(Number)
+    }
+
+    const slices = Object.values(this.images)
+    const meta = slices[0].metaData
+
+    // Origin
+    let origin = [0, 0, 0]
+    if (meta.ImagePositionPatient !== undefined) {
+      origin = numArrayFromString(meta.ImagePositionPatient)
+    }
+
+    // Spacing
+    let spacing = [1, 1];
+    if (meta.PixelSpacing !== undefined) {
+      spacing = numArrayFromString(meta.PixelSpacing);
+    }
+    if (meta.SliceThickness !== undefined) {
+      spacing.push(Number(meta.SliceThickness))
+    } else {
+      spacing.push(1);
+    }
+
+    // Dimensions
+    const size = [
+      meta.Columns,
+      meta.Rows,
+      Object.keys(this.images).length,
+    ]
+
+    // Direction matrix (3x3)
+    let directionCosines = [1, 0, 0, 0, 1, 0]
+    if (meta.ImageOrientationPatient !== undefined) {
+      directionCosines = numArrayFromString(meta.ImageOrientationPatient)
+    }
+    const iDirCos = directionCosines.slice(0, 3)
+    const jDirCos = directionCosines.slice(3, 6)
+    const kDirCos = [
+      iDirCos[1] * jDirCos[2] - iDirCos[2] * jDirCos[1],
+      iDirCos[2] * jDirCos[0] - iDirCos[0] * jDirCos[2],
+      iDirCos[0] * jDirCos[1] - iDirCos[1] * jDirCos[0],
+    ]
+    const direction = {
+      rows: 3,
+      columns: 3,
+      data: [
```

Comment:
For me, parsing the same tags and using this code, direction.data ended up being: [...iDirCos, ...jDirCos, ...kDirCos]. Just a heads up!